github_connector/models/github_repository_branch.py

```diff
@@ -275,8 +291,65 @@ def _analyze_code(self):
                         branch.name,
                         e,
                     )
+
         return True
 
+    def _prepare_analysis_rule_model_info(self, analysis_rule_id):
+        """
+        Define model data info that override with other addons
+        """
+        return "github.repository.branch.rule.info"
+
+    def _delete_analysis_rule_model_info(self, analysis_rule_id):
+        """
+        Remove existing info data to create new records
+        """
+        self.env[self._prepare_analysis_rule_model_info(analysis_rule_id)].search(
+            [
+                ("analysis_rule_id", "=", analysis_rule_id.id),
+                ("repository_branch_id", "=", self.id),
+            ]
+        ).sudo().unlink()
+
+    def _prepare_analysis_rule_info_vals(self, analysis_rule_id):
+        """
+        Prepare info vals
+        """
+        res = self._operation_analysis_rule_id(analysis_rule_id)
+        return [
+            {
+                "analysis_rule_id": analysis_rule_id.id,
+                "repository_branch_id": self.id,
+                "code_count": res["code"],
+                "documentation_count": res["documentation"],
+                "empty_count": res["empty"],
+                "string_count": res["string"],
+                "scanned_files": len(res["paths"]),
+            }
+        ]
+
+    def _operation_analysis_rule_id(self, analysis_rule_id):
+        """
+        This function allow to override with other addons that need
+        to change this analysis
+        """
```

Comment:
Multi-line docstring is formatted this way:
```suggestion
        """This function allows to be overriden by other addons that need
        to change the analysis.
        """
```